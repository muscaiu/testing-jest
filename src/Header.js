import React, { Component } from 'react';

class Header extends Component {
    constructor(props) {
        super(props)
        this.state = {
            name: 'state in Header'
        }
    }

    render() {
        return (
            <p>{this.state.name}</p>
        )
    }
}

export default Header
