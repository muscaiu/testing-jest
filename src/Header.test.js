import React from 'react';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Header from './Header'

it('shallow renders Header', () => {
    shallow(<Header />)
})

it('renders <Header/> snapshot', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  