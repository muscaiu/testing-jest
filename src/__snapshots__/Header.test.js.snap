// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Header/> snapshot 1`] = `
<div>
  Header
</div>
`;
